HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponentsController+Helpers.swift

```diff
@@ -0,0 +1,806 @@
+            switch result {
+            case .success(let data):
+                completion(.success(data))
+            case .failure(let error):
+                let healthError = GiniHealthAPILibrary.GiniError.unknown(response: error.response, data: error.data)
+                completion(.failure(healthError))
+            }
+        }
+    }
+    
+    /**
+     Opens the payment provider app using the specified request ID and universal link.
+
+     - Parameters:
+       - requestId: The ID of the payment request.
+       - universalLink: The universal link to open the payment provider app.
+     */
+    public func openPaymentProviderApp(requestId: String, universalLink: String) {
+        giniSDK.openPaymentProviderApp(requestID: requestId, universalLink: universalLink)
+    }
+    
+    /**
+        Retrieves a payment request using the provided payment request ID.
+
+        - Parameter id:         The ID of the payment request to retrieve.
+        - Parameter completion: A closure to be executed once the retrieval is completed, containing the result of the operation as a `PaymentRequest` object on success or a `GiniError` on failure.
+    */
+    public func getPaymentRequest(by id: String, completion: @escaping (Result<PaymentRequest, GiniHealthAPILibrary.GiniError>) -> Void) {
+        giniSDK.getPaymentRequest(by: id) { result in
+            switch result {
+            case .success(let paymentRequest):
+                completion(.success(paymentRequest))
+            case .failure(let error):
+                let healthError = GiniHealthAPILibrary.GiniError.unknown(response: error.response, data: error.data)
+                completion(.failure(healthError))
+            }
+        }
+    }
+}
+
+extension PaymentComponentsController: BanksSelectionProtocol {
+    /// Updates the selected payment provider and notifies the delegate with the provider and optional document ID.
+    public func didSelectPaymentProvider(paymentProvider: GiniHealthAPILibrary.PaymentProvider) {
+        selectedPaymentProvider = PaymentProvider(healthPaymentProvider: paymentProvider)
+        if let provider = selectedPaymentProvider {
+            storeDefaultPaymentProvider(paymentProvider: provider)
+            self.presentPaymentViewBottomSheet()
+        }
+    }
+
+    /// Handles the action when the continue button is tapped on the share bottom sheet.
+    public func didTapOnContinueOnShareBottomSheet() {
+        GiniUtilites.Log("Tapped Continue on Share Bottom Sheet", event: .success)
+    }
+
+    /// Handles the action when the forward button is tapped on the install bottom sheet.
+    public func didTapForwardOnInstallBottomSheet() {
+        GiniUtilites.Log("Tapped Forward on Install Bottom Sheet", event: .success)
+    }
+
+    /// Handles the action when the pay button is tapped on install bottom sheet.
+    public func didTapOnPayButton() {
+        //
```

Comment:
[clarification] Nothing to do here?

Reply:
It's done only on MerchantSDK, this is due to common GiniInternalPaymentSDK protocols. I'll remove the comments